packages/kit/src/runtime/client/client.js

```diff
@@ -534,7 +536,7 @@ async function get_navigation_result_from_branch({
 			route: {
 				id: route?.id ?? null
 			},
-			state: {},
+			state: state ?? {},
```

Comment:
```suggestion
			state: {},
```
I think we want to reset `state` when we navigate similar to the form logic.
`Reset `form` on navigation, but not invalidation`

Reply:
That would leave the entire PR useless and the test should fail.
We want to keep the current state when invalidating.
The `state` prop is only passed to this fn in line 998 and only if `invalidating` is true.

Reply:
Ah you're right. I think I understand now. Maybe Simon meant  something like the below in combination with the default `state` argument value:
```suggestion
			state,
```
But I think the way you've written it is equivalent so it should be fine unless I missed something.